examples/inertia_relief/inertia_relief_example.cpp

```diff
@@ -312,31 +308,33 @@ int main(int argc, char* argv[])
     writer.write(0, 0.0, objective_states);
   }
   auto non_const_states = getFieldPointers(states, params);
+  // create an inertial relief problem
   InertialReliefProblem problem({non_const_states[DISP], non_const_states[DENSITY]}, non_const_states, mesh,
                                 solid_mechanics_weak_form, constraints);
-  int dimx = problem.GetDimx();
-  int dimy = problem.GetDimy();
-
-  mfem::Vector x0(dimx);
-  x0 = 0.0;
-  mfem::Vector y0(dimy);
-  y0 = 0.0;
-  mfem::Vector xf(dimx);
-  xf = 0.0;
-  mfem::Vector yf(dimy);
-  yf = 0.0;
 
+  // optimization variables
+  auto X0 = problem.GetOptimizationVariable();
+  auto Xf = problem.GetOptimizationVariable();
+
+  // define a homotopy solver for the inertia relief problem
   HomotopySolver solver(&problem);
+  // set solver options
   solver.SetTol(nonlinear_absolute_tol);
   solver.SetMaxIter(nonlinear_max_iterations);
 
-  solver.Mult(x0, y0, xf, yf);
+  // solve the inertia relief problem
+  solver.Mult(X0, Xf);
+  // extract displacement and Lagrange multipliers
+  mfem::Vector displacement_sol = problem.GetDisplacement(Xf);
+  mfem::Vector multiplier_sol = problem.GetLagrangeMultiplier(Xf);
   bool converged = solver.GetConverged();
   if (myid == 0) {
     if (converged) {
-      std::cout << "converged!\n";
+      std::cout << "Converged!\n";
```

Comment:
Can you swap these to the appropriate SLIC calls? That way any global logger changes affect these.